src/backend/session/auto.rs

```diff
@@ -61,11 +66,47 @@ pub enum AutoSessionNotifier {
     Logind(LogindSessionNotifier),
     /// Direct / tty session notifier
     Direct(DirectSessionNotifier),
+    /// LibSeat session notifier
+    #[cfg(feature = "backend_session_libseat")]
+    LibSeat(LibSeatSessionNotifier),
 }
 
 impl AutoSession {
     /// Tries to create a new session via the best available interface.
-    #[cfg(feature = "backend_session_logind")]
+    #[cfg(feature = "backend_session_libseat")]
+    pub fn new<L>(logger: L) -> Option<(AutoSession, AutoSessionNotifier)>
```

Comment:
Maybe we can flatten all this into a single method that does early return with internal `#[cfg(..)]` blocks, instead of having 3 different `new()` methods, that would probably be easier to follow and reduce code duplication.

Something like this:

```rust
pub fn new<L>(logger: L) -> Option<(AutoSession, AutoSessionNotifier)> {
    #[cfg(feature = "backend_session_libseat")]
    {
        /* try to init libseat backend, early return on success */
    }
    #[cfg(feature = "backend_session_logind")]
    {
        /* try to init logind backend, early return on success */
    }
    /* try to init direct backend */
}
```

Reply:
https://github.com/Smithay/smithay/pull/292/commits/4832951d0e1ed4900ea7d53c0d8af97839501884
Indeed, looks a lot cleaner.